rllib/algorithms/algorithm.py

```diff
@@ -2131,6 +2148,17 @@ def load_checkpoint(self, checkpoint: Union[Dict, str]) -> None:
         else:
             checkpoint_data = checkpoint
         self.__setstate__(checkpoint_data)
+        if isinstance(checkpoint, str) and self.config._enable_learner_api:
```

Comment:
I don't think the location of this logic ties well to the existing code where checkpoint can take both a dict or str value. You need to map the checkpoint input (str or dict) to a checkpoint data first and then use checkpoint data inside the `__setstate__()` api to set the state of the learner group.

Reply:
my question is more like what do you need to do if checkpoint is a dict? when would that happen, and what would that mean for the learner group

Reply:
ok so this is interesting. upon further inspection, this reason that this is supposed to accept a dict is in the case that `trainable.save_checkpoint` ever returns a dictionary. However, we don't do this, which means that we don't need to even support this inside of `load_checkpoint` to begin with. I just ended up removing all the logic related to handling dicts.